src/tools/tests/storage-key-recipe-resolver-test.ts

```diff
@@ -28,33 +29,147 @@ describe('recipe2plan', () => {
     
     @trigger
       launch startup
-      arcId myArcId
+      arcId writeArcId
     recipe WritingRecipe
       thing: create persistent 'my-handle-id' 
       Writer
         data: writes thing
 
     @trigger
       launch startup
-      arcId otherArcId
+      arcId readArcId
     recipe ReadingRecipe
       data: map 'my-handle-id'
       Reader
         data: reads data`);
 
       const resolver = new StorageKeyRecipeResolver(manifest);
-      for  (const it of (await resolver.resolve())) {
+      for (const it of (await resolver.resolve())) {
         assert.isTrue(it.isResolved());
       }
     });
+    it('fails to resolve mapping a handle from a short running arc into another short running arc', async () => {
+      const manifest = await Manifest.parse(`\
+    particle Reader
+      data: reads Thing {name: Text}
+
+    particle Writer
+       data: writes Thing {name: Text}
+    
+    recipe WritingRecipe
+      thing: create persistent 'my-handle-id' 
+      Writer
+        data: writes thing
+
+    recipe ReadingRecipe
+      data: map 'my-handle-id'
+      Reader
+        data: reads data`);
+
+
+      const resolver = new StorageKeyRecipeResolver(manifest);
+      await assertThrowsAsync(async () => {
+        for (const it of await resolver.resolve()) {
```

Comment:
Why iterate over the result? Why not just `await resolver.resolve();` ?

Reply:
Ah. This is vestigial from when I was using generators. Thanks for your suggestion.